### PR TITLE
Don't allow users to view Collections as Projects [OSF-5843]

### PR DESCRIPTION
**Pull request by erinspace**
https://openscience.atlassian.net/browse/OSF-5843

# Purpose
Right now, users can access a collection by visiting it's guid as if it were a normal project. This PR adds a check to the ```@must_be_valid_project``` decorator that checks to see if it's a collection, and throws a 404 error with relevant message if it is.

# Changes
- Add check to ```@must_be_valid_project``` to see if guid being accessed is a collection

# Side Effects
None I can think of

### CODE REVIEW

File: website/project/decorators.py [hunk 1]
```diff
@@ -86,6 +86,12 @@ def wrapped(*args, **kwargs):
 
             _inject_nodes(kwargs)
 
+            if getattr(kwargs['node'], 'is_collection', True):
+                raise HTTPError(
+                    http.BAD_REQUEST,
```
Comment (mfraezz) [inline website/project/decorators.py]: As per the Jira ticket, this should be a `404 NOT FOUND`

**Comment by mfraezz** [comment]
Minor inconsistency with the expected behavior described on the ticket. Otherwise, LGTM. :fire:

File: website/project/decorators.py (hunk 1 above)
Comment (brianjgeiger) [inline website/project/decorators.py]: There's a similar error with retracted registrations that treated the same, so we were leaning towards matching that.

Comment (brianjgeiger) [inline website/project/decorators.py]: It'll probably be quick to change either way, though, if retractions feel significantly different from this.

**Comment by mfraezz** [comment]
Changes look good, dropping this in RTM :pencil2:

**Pull request merged**